nixos/modules/system/vars/on-machine-backend.nix

```
@@ -0,0 +1,140 @@
+# we use this vars backend as an example backend.
+# this generates a script which creates the values at the expected path.
+# this script has to be run manually (I guess after updating the system) to generate the required vars.
+{
+  pkgs,
+  lib,
+  config,
+  ...
+}:
+let
+  cfg = config.vars.settings.on-machine;
+  sortedGenerators =
+    (lib.toposort (a: b: builtins.elem a.name b.dependencies) (lib.attrValues config.vars.generators))
+    .result;
+
+  promptCmd = {
+    hidden = "read -sr prompt_value";
+    line = "read -r prompt_value";
+    multiline = ''
+      echo 'press control-d to finish'
+      prompt_value=$(cat)
+    '';
+  };
+  generate-vars = pkgs.writeShellApplication {
```

If I understand correctly, this script may be run on the deployer host, which can be a different platform.

```suggestion
  generate-vars = deployerPkgs: deployerPkgs.writeShellApplication {
```

It is then up to the caller to pass a suitable `pkgs` instance that works for the deployer host.

---

As a test case, you could add `aarch64-darwin` to [this](https://github.com/Lassulus/vars/blob/066dbc738838b79be6bc80b88559cb88e0fffffd/flake.nix#L9-L14) and run the `checks` on darwin.

---

usually we overwrite pkgs of the system where the vars are defined if the caller architecture is different from the target systems architecture, since this also supports string context references to pkgs and the vars definitons want to live next to the service definitions.